### PR TITLE
Fix #87, Change EVS_Register failure from SendEvent to WriteToSysLog

### LINKED ISSUE #87: FM tries to send an event message on failure of registering with EVS
**Checklist**
* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I performed a cursory search to see if the bug report is relevant, not redundant, nor in conflict with other tickets.

**Describe the bug**
Copy of https://github.com/nasa/DS/issues/85
FM and DS are the only apps to (attempt to) utilize `CFE_EVS_SendEvent` even though the call to `CFE_EVS_Register` failed.
Better update to `CFE_ES_WriteToSysLog` so failure can be recorded somewhere. This brings FM (and DS) in line with the other cFS apps.

**Code snips**
https://github.com/nasa/FM/blob/2be3511211e756058236b30abe40b0de372bd2f1/fsw/src/fm_app.c#L175-L182

**Expected behavior**
`CFE_ES_WriteToSysLog` on failure.

**Reporter Info**
Avi Weiss @thnkslprpt

### PR DESCRIPTION
**Pull request by thnkslprpt**
**Checklist**
* [x] I reviewed the [Contributing Guide](https://github.com/nasa/osal/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate [Contributor License Agreement](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Fixes #87
`CFE_EVS_SendEvent` replaced by `CFE_ES_WriteToSysLog` on failure of `CFE_EVS_Register`.

**Testing performed**
GitHub CI actions (incl. Build + Run, Unit Tests etc.) all passing successfully.
Coverage maintained at 100%:
```
  lines......: 100.0% (1443 of 1443 lines)
  functions..: 100.0% (72 of 72 functions)
  branches...: 100.0% (525 of 525 branches)
```

**Expected behavior changes**
Failure can be successfully recorded somewhere even without the EVS now.

**Contributor Info**
Avi Weiss @thnkslprpt

### CODE REVIEW

File: unit-test/fm_app_tests.c [hunk 1]
```diff
@@ -242,9 +243,11 @@ void Test_FM_AppInit_EVSRegisterNotSuccess(void)
...
     /* Assert */
     UtAssert_STUB_COUNT(CFE_EVS_Register, 1);
-    UtAssert_STUB_COUNT(CFE_EVS_SendEvent, 1);
-    UtAssert_INT32_EQ(context_CFE_EVS_SendEvent[0].EventID, FM_STARTUP_EVENTS_ERR_EID);
-    UtAssert_INT32_EQ(context_CFE_EVS_SendEvent[0].EventType, CFE_EVS_EventType_ERROR);
+    UtAssert_STUB_COUNT(CFE_ES_WriteToSysLog, 1);
+
+    call_count_CFE_EVS_SendEvent = UT_GetStubCount(UT_KEY(CFE_EVS_SendEvent));
+    UtAssert_True(call_count_CFE_EVS_SendEvent == 0, "CFE_EVS_SendEvent was called %u time(s), expected 0",
+                  call_count_CFE_EVS_SendEvent);
```
Comment (jphickey) [inline unit-test/fm_app_tests.c]: I would recommend taking out these two assertions, no real need to assert that a function was not called.  Also no need to add the global variable either.

**Review by jphickey — CHANGES_REQUESTED** [review]
Minor update requested, but otherwise looks good!  Thanks.

File: unit-test/fm_app_tests.c (hunk 1 above)
Comment (thnkslprpt) [reply unit-test/fm_app_tests.c]: No worries @jphickey 
Removed those two.

**Review by jphickey — APPROVED** [review]
Looks good, thanks!

**Pull request merged**